### PR TITLE
Add MTEBEvaluator for embedding model evaluation

### PR DESCRIPTION
**Pull request by natke**
## Summary

Add a built-in MTEB (Massive Text Embedding Benchmark) evaluator to Olive, following the same architecture as `LMEvaluator`/`lmeval_ort.py`.

### `olive/evaluator/olive_evaluator.py`
- New `MTEBEvaluator` class registered in the evaluator registry
- Supports three model classes: `hf` (SentenceTransformer), `ort` (ONNX via ORT), `ortgenai` (GenAI with hidden_states)
- Auto-detects model class from handler type, including GenAI detection via `genai_config.json`

### `olive/evaluator/mteb_ort.py` (new)
- `MTEBOnnxBase`: abstract base implementing MTEB's EncoderProtocol with mean pooling
- `MTEBORTEvaluator`: wraps plain ONNX models via `ort.InferenceSession`
- `MTEBORTGenAIEvaluator`: wraps GenAI models using `og.Generator` + `hidden_states` output

## Usage

```json
"evaluators": {
    "mteb": {
        "type": "MTEBEvaluator",
        "tasks": ["STS17"],
        "batch_size": 32
    }
},
"evaluator": "mteb"
```

## Testing

Tested end-to-end with Qwen3-Embedding-0.6B CPU recipe — ModelBuilder export + MTEB STS17 evaluation on both input HF model and exported GenAI model.

### CODE REVIEW

File: olive/evaluator/mteb_ort.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        for batch in inputs:
+            if isinstance(batch, dict) and "text" in batch:
+                sentences.extend(batch["text"])
+            elif isinstance(batch, (list, tuple)):
+                sentences.extend(batch)
+            elif isinstance(batch, str):
+                sentences.append(batch)
+            else:
+                sentences.extend(list(batch))
+
```
[the remark this answers is not in the document]
Comment (natke) [reply olive/evaluator/mteb_ort.py:73]: Fixed in e6f3e23 — added `isinstance(inputs, str)` guard before the iteration loop to wrap single strings in a list.

File: olive/evaluator/mteb_ort.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            logger.debug("hidden_states output not available, trying logits-based approach")
+
+        # Fallback: get logits and use them as-is (not ideal for embeddings)
+        logits = generator.get_output("logits")
+        logits = np.array(logits, copy=False)
+        if logits.ndim == 2:
+            embed_dim = logits.shape[-1]
+            logits = logits.reshape(batch_size, -1, embed_dim)
+        return self._mean_pool(logits, attention_mask)
+
```
[the remark this answers is not in the document]
Comment (natke) [reply olive/evaluator/mteb_ort.py:256]: Fixed in e6f3e23 — removed the logits fallback entirely. Now raises `RuntimeError` with a clear message asking the user to ensure the model was built with `include_hidden_states=1`.

**Review by jambayk — APPROVED** [review]
